lib/fs.js

```diff
@@ -783,6 +783,10 @@ fs.renameSync = function(oldPath, newPath) {
 
 fs.truncate = function(path, len, callback) {
   if (typeof path === 'number') {
+    process.emitWarning(
```

Comment:
Please use `util.deprecate()` for runtime deprecations. See how it's done [here](https://github.com/nodejs/node/blob/f6caeb952605ffa9968ccd88eb330829f980a536/lib/os.js#L158).

Reply:
I asked one of the mentor during the event, and looks like for this case since we are deprecating a particular usage, we cannot use util.deprecate().

Reply:
Indeed. But we still need to make sure this warning is only emitted once and that it respects `--no-deprecation`. See how it's done with Buffer [here](https://github.com/nodejs/node/blob/master/lib/buffer.js#L163) (although that's a pending deprecation for now, so make sure to use `process.noDeprecation` instead of `pendingDeprecation`)

Reply:
Oh I see what you mean, thank you for sharing the resources, I will make changes according to what you said.